### DISCUSSION TITLE
v7.2.0 Release of PowerShell

### DISCUSSION THREAD

**Discussion by adityapatwardhan** [PowerShell/PowerShell#16396] [General]
### General Cmdlet Updates and Fixes

- Handle exception when trying to resolve a possible link path (#16310)

### Tests

- Fix global tool and SDK tests in release pipeline (#16342)

### Build and Packaging Improvements

<details>

<summary>

<p>We thank the following contributors!</p>
<p>@kondratyev-nv</p>

</summary>

<ul>
<li>Add an approval for releasing build-info json (#16351)</li>
<li>Release build info json when it is preview (#16335)</li>
<li>Update <code>metadata.json</code> for v7.2.0 release</li>
<li>Update to the latest notices file and update <code>cgmanifest.json</code> (#16339)(#16325)</li>
<li>Fix issues in release build by updating usage of <code>powershell.exe</code> with <code>pwsh.exe</code> (#16332)</li>
<li>Update feed and analyzer dependency (#16327)</li>
<li>Update to .NET 6 GA build <code>6.0.100-rtm.21527.11</code> (#16309)</li>
<li>Add a major-minor build info JSON file (#16301)</li>
<li>Fix Windows build ZIP packaging (#16299) (Thanks @kondratyev-nv!)</li>
<li>Clean up <code>crossgen</code> related build scripts also generate native symbols for <code>R2R</code> images (#16297)</li>
<li>Fix issues reported by code signing verification tool (#16291)</li>
</ul>

</details>

[7.2.0]: https://github.com/PowerShell/PowerShell/compare/v7.2.0-rc.1...v7.2.0

### SHA256 Hashes of the release artifacts

- hashes.sha256
  - 742214B9DD9805522F1ACF16BF7265BAC6CAC06BC669CBFC877B514279D3D88C
- powershell_7.2.0-1.deb_amd64.deb
  - 9FCDE25BBD7F537213E8AE34A9803C1DED59CFE1D41701816C424F9B33FD2BB3
- powershell-7.2.0-1.rh.x86_64.rpm
  - A1F78D1807A79C691256434E7E913F7044D32A535222F1305CB408E0EE0065F9
- powershell-7.2.0-linux-alpine-x64.tar.gz
  - DCCF213728B6D539785CBCC9AD9BC0971D82C7E8CF010E8F949E3DD17FC4C596
- powershell-7.2.0-linux-arm32.tar.gz
  - BB1E2CA5FF230721ED3D2579714A7CFE12EA27B7B27B22AE222B3515D9038B8C
- powershell-7.2.0-linux-arm64.tar.gz
  - 19E9AA2A308A68E6CF90AD11EBA86948EF031850E00B0D0F9E26AC9D59E46A24
- powershell-7.2.0-linux-x64.tar.gz
  - 15B67D0571D9891A10E541A1462C906084AB148614DD83C6BBDDF90046A815A8
- powershell-7.2.0-linux-x64-fxdependent.tar.gz
  - 6C842AB8EC48553E4C246CE8A01FCC19E9173726674C902B5ABAF4A4A1B0D0B0
- powershell-7.2.0-osx-arm64.pkg
  - 6293A48CEAD807DD3C9F932176021533254BD6883754C5574E36A1775D797959
- powershell-7.2.0-osx-arm64.tar.gz
  - B48A07970550FE506B91EC2FA009FC8742B9DC20CACC2E34A6FE8A8F588149F1
- powershell-7.2.0-osx-x64.pkg
  - F73682927BDADB127CF0C18641D5EFE9EA12E53053E65D5C54F06FD4EF413A4A
- powershell-7.2.0-osx-x64.tar.gz
  - B1864C204C7861FF8416446335FA002C5E48D2D50292B6E221E05C946AA8B9AE
- PowerShell-7.2.0-win-arm32.zip
  - 5740E5BAD2E2284AF336DD09557FA752902D1C5CF991F5F909CAD163CC0EBB8A
- PowerShell-7.2.0-win-arm64.zip
  - 9E46EEF382B5C36E25BC64D8275183A1A58EC6630D02B6F1636900F11D8D0A54
- PowerShell-7.2.0-win-fxdependent.zip
  - 1752A8389188F6B792E5C1AE17E85EA0E48EE3908A7C3BF02BAEC39132500356
- PowerShell-7.2.0-win-fxdependentWinDesktop.zip
  - 75C1DC94B5DF666815731802E1BFEC936A996B20561BB1DF5A1292082E767971
- PowerShell-7.2.0-win-x64.msi
  - BD5BDDF2FAC4626479C4F381BDB30A0DC6A9516BE2DE9FC726F33C65F56FE9D4
- PowerShell-7.2.0-win-x64.zip
  - 0D38173C440CFE5FC123E82003FA5C68CED0AA1769619658364425A110F2016D
- PowerShell-7.2.0-win-x86.msi
  - BA8C9DA25E127E8B8712C20E3C05503F21AB6505ECA6042743E9C3898DF22411
- PowerShell-7.2.0-win-x86.zip
  - 2522D078E6457052BBF8387AFD2FA186486F4905BF0BED4C70AB58AA6E6B4814
- powershell-lts_7.2.0-1.deb_amd64.deb
  - C2A689ACAF91B5FAAA5A57306B4BCDED271C6C87BE0DDD08DB6CBA98097CBD9B
- powershell-lts-7.2.0-1.rh.x86_64.rpm
  - C2AE8BC3B2C3742A2C6EA98A012CF16EEFD39764E27A0F88712476E97A1295E5
- powershell-lts-7.2.0-osx-arm64.pkg
  - 473C8D5FEC4B90E6BEAB2ABB0DF29B2447D7CBBC666D4413DDBC7CA41D2A4E43
- powershell-lts-7.2.0-osx-x64.pkg
  - 64436E6A927C6F1026272EAC00F95D15EBBCDB08475707965576907A26650CB7

<hr /><em>This discussion was created from the release <a href='https://github.com/PowerShell/PowerShell/releases/tag/v7.2.0'>v7.2.0 Release of PowerShell</a>.</em>

**Comment by JustinGrote** [comment]
Congrats to the team on another LTS release!

**Comment by doctordns** [comment]
Awesome!!

**Comment by mattcargile** [comment]
I don't see the `winget` release up yet. Is it still appropriate to use this channel to upgrade?

**Reply by JustinGrote** [reply]
I imagine it will be working its way to distro channels over the course of this week, however with the MSI version it can now use Microsoft Update to keep up to date so you no longer need to use winget.

**Reply by mattcargile** [reply]
Thanks, Justin. I wonder how it will work on a corporate network with Windows Update disabled. I guess I would still have to initiate the update myself.

EDIT: I'm assuming if my `wt.exe` is staying up-to-date then this would follow the same pattern?

**Reply by iSazonov** [reply]
>  I wonder how it will work on a corporate network with Windows Update disabled

WSUS, SCCM :-)

> I'm assuming if my wt.exe is staying up-to-date then this would follow the same pattern?

pwsh is released as Appx too so you can install pwsh from Store too.

**Reply by xiaolongwuhpu** [reply]
How do I upgrade in the PowerShell panel ?
I don't want to download zip or msi !!!

**Comment by xiaolongwuhpu** [comment]
How do I upgrade in the PowerShell panel ?
I don't want to download zip or msi !!!

**Reply by JustinGrote** [reply]
How do you mean "in the Powershell panel", do you mean from the Powershell CLI? Are you on Windows or Linux? There are many scripts/tools to install depending.

**Reply by enderandpeter** [reply]
Fortunately, the MSI has added an option to keep the program up to date via Windows Update!

**Comment by YetaEdgarMutelo** [comment]
When is it gonna be functional with Exchange Online and Azure AD?

**Reply by iSazonov** [reply]
Ask Exchange Online and Azure AD teams :-)

**Reply by JustinGrote** [reply]
That said, Azure AD works fine via the Powershell Graph SDK (and on Windows with the WindowsCompatibility module), and Exchange Online module works just fine with v2 cross-platform.

**Comment by sebademasi10** [comment]
Nice! I am learning to write scripts to automate my developer tasks 😄 
I have a question: is there a way to upgrade from 7.1 using powershell itself? 🤔

**Reply by mattcargile** [reply]
`winget upgrade --exact --id Microsoft.PowerShell` from `cmd.exe` or `powershell.exe`. It isn't strictly PowerShell but it is a command line tool. `choco` could also be used most likely.

**Reply by JustinGrote** [reply]
There is also this script: 
https://github.com/PowerShell/PowerShell/blob/master/tools/install-powershell.ps1

**Reply by sebademasi10** [reply]
I've just installed it with `winget`. Thanks to both!

**Comment by nimanch** [comment]
Hi!, 
I noticed that ubuntu18.04 and ubuntu20.04 packages are missing from this release. What should we use for each of those OS ? Please advise?

**Reply by JustinGrote** [reply]
There's a universal package for all versions of ubuntu now:

https://docs.microsoft.com/en-us/powershell/scripting/install/install-ubuntu?view=powershell-7.2#installation-via-direct-download
https://github.com/PowerShell/PowerShell/releases/download/v7.2.0/powershell-lts_7.2.0-1.deb_amd64.deb

However it doesn't appear to be publishing to the apt repositories. @adityapatwardhan can inspect?

**Reply by kilasuit** [reply]
@JustinGrote I could install via APT on a brand new Ubuntu 20.04 easily enough by following this https://docs.microsoft.com/en-us/powershell/scripting/install/install-ubuntu?view=powershell-7.2#installation-via-package-repository

**Comment by rjmholt** [comment]
> powershell-7.2.0-osx-arm64.pkg

Very cool

**Comment by rhubarb-geek-nz** [comment]
Could you create RPM and Debian packages for the Linux arm32 and arm64 please? This would make it much more friendly to install on a Raspberry Pi.
I have been creating these manually, but it makes sense for this project to create them at the same time as the x86_64 packages.
These could then also be signed by your official key.

See [rhubarb pi - pkg-powershell](https://sourceforge.net/p/rhubarb-pi/wiki/pkg-powershell/)

Thanks

**Reply by kilasuit** [reply]
@rhubarb-geek-nz  Please raise this as an Issue using the Ditsribution Request template for consideration

**Reply by rhubarb-geek-nz** [reply]
I had a look at that template and that would require me to create a new distribution request for every release for each architecture. My suggestion was to create the arm32 and arm64 rpm and deb in the same manner as you create the x86_64 ones and at the same time when you´create a new release.

**Comment by nmbell** [comment]
`Get-Help -Online` opens the 7.1 help pages instead of the 7.2 pages. Is this a PowerShell issue or a web site issue?

**Reply by JustinGrote** [reply]
A bit of both. The -Online links tie to fwlinks, here's the one for get-command
https://go.microsoft.com/fwlink/?LinkID=2096579
The key part is `view=powershell-7`

The links should be updated to 7.2, sound like a good candiate for a 7.2.1 servicing release @adityapatwardhan :).

@nmbell can you open an issue if one doesn't exist?

**Reply by nmbell** [reply]
Done: #16479

**Reply by adityapatwardhan** [reply]
Yes this is known. It is scheduled for 7.2.1 @JustinGrote said.

Thanks for reporting.

**Comment by sebademasi10** [comment]
Hi! Is there a way to resume a stopped job? The Microsoft documentation suggest to use `Resume-Job` but when I do it I get the following error: 
```powershell
PS > PS C:\Program Files\PowerShell\7> Resume-Job
Resume-Job: The term 'Resume-Job' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```
Thanks in advance 😌

**Reply by JustinGrote** [reply]
A stopped job is not a "paused" job, you just have to start a new one.

FYI this is for discussion of issues with the 7.2 release specifically. If you need general help I highly recommend aka.ms/PSDiscord, there's a great community there happy to help.

**Comment by sebademasi10** [comment]
Hi again! I love pwsh <3

I'd needing a way to change the location where pwsh starts, is it possible?

Thanks in advance! 🤗

**Reply by rhubarb-geek-nz** [reply]
Do you mean set what the current directory is?

~~~
-WorkingDirectory | -wd

    Sets the initial working directory by executing at startup. Any valid
    PowerShell file path is supported.

    To start PowerShell in your home directory, use: pwsh -WorkingDirectory ~
~~~

**Reply by sebademasi10** [reply]
Yep, but I think your solution is to execute pwsh from another pwsh. I need to change the initial working directory, for example, when I execute pwsh from home button: 

![image](https://user-images.githubusercontent.com/53543566/143490889-6edcc2ed-010a-434e-841c-1e0114a69fb2.png)

**Reply by rhubarb-geek-nz** [reply]
Open file location of the start item, select the item and choose "properties" then change "Start in" field

**Reply by sebademasi10** [reply]
Nice! It worked! I remembered that way but I was trying to do it with the shortcut. Thanks!!!

**Reply by kilasuit** [reply]
@sebademasi10 in future please start new discussions instead of adding to an existing one with a question that's not relevant to the ongoing discussion

**Reply by JustinGrote** [reply]
@sebademasi10 check out aka.ms/PSDiscord for general non 7.2 related questions, people there are very helpful and respond much quicker!
